imaging/tools/main.ts

```diff
@@ -85,36 +85,22 @@ const csToolsCreateStack = function (
   cornerstoneTools.addToolState(element, "stack", stack);
 };
 
-export function csToolsUpdateImageIds(
-  elementId: string,
-  imageIds: string[],
-  imageIdIndex: number
-) {
+/**
+ * Update stack object to sync stack tools
+ * @function csToolsUpdateImageIds
+ * @param {String} elementId - The target html element id.
+ * @param {Array} imageIds - Stack image ids.
+ */
+
+export function csToolsUpdateImageIds(elementId: string, imageIds: string[]) {
   const element = document.getElementById(elementId);
   if (element) {
     const stackState = cornerstoneTools.getToolState(element, "stack");
     const stackData = stackState.data[0];
     stackData.imageIds = imageIds;
-    stackData.currentImageIdIndex =
```

Comment:
questo deve rimanere ma deve essere semplicemente `=imageIdIndex`

Reply:
Noooooooooooo! quando hai voglia ne parliamo..

Reply:
penso lo faccia già automaticamente di fare l'update del currentImageIdIndex in updateStackToolState (come store.get(["viewports", element.id, "sliceId"]);), quindi la riga non è necessaria

Reply:
Esatto, la riga non è necessaria ed è anche concettualmente sbagliata. Questa funzione è usata e chiamata solo per aggiornare l'array degli ids durante il caricamento progressivo, l'index è già aggiornato correttamente durante il rendering